langserver/completion.go

```diff
@@ -71,9 +74,45 @@ func (h *LangHandler) handleTextDocumentCompletion(ctx context.Context, conn jso
 				NewText: it.Name,
 			},
 		}
+		item = h.addInsertText(item)
+		citems[i] = item
 	}
 	return &lsp.CompletionList{
 		IsIncomplete: false,
 		Items:        citems,
 	}, nil
 }
+
+func (h *LangHandler) addInsertText(item lsp.CompletionItem) lsp.CompletionItem {
+    item.InsertTextFormat = lsp.ITFPlainText
+	if item.Kind == lsp.CIKFunction {
+		args := parseFuncArgs(item.Detail)
+        suffix := ""
+		if h.init.Capabilities.TextDocument.Completion.CompletionItem.SnippetSupport {
+			item.InsertTextFormat = lsp.ITFSnippet
+			args = genSnippetArgs(args)
+            suffix = "$0"
+		}
+		item.InsertText = fmt.Sprintf("%s(%s)%s", item.Label, strings.Join(args, ", "), suffix)
```

Comment:
`CompletionItem.InsertText` is deprecated. Simply override the value of `CompletionItem.TextEdit[0].NewText` here, which also applies the format specified in `CompletionItem.InsertTextFormat`.

Reply:
Please do not modify the completion text if `SnippetSupport` is false. It would be annoying to delete the arguments snippet hint manually.

Reply:
Agreed.  I should have these changes done later today.